libgnucash/engine/gnc-pricedb.cpp

```diff
@@ -2346,26 +2337,17 @@ gnc_pricedb_lookup_nearest_before_t64 (GNCPriceDB *db,
                                        time64 t)
 {
     GNCPrice *current_price = NULL;
-
     if (!db || !c || !currency) return NULL;
     ENTER ("db=%p commodity=%p currency=%p", db, c, currency);
-
     auto price_list = pricedb_get_prices_internal (db, c, currency, TRUE);
     if (!price_list) return NULL;
-
-    for (auto item = price_list; item; item = item->next)
+    auto p = g_list_find_custom (price_list, GUINT_TO_POINTER(t), (GCompareFunc)price_time64_less_or_equal);
```

Comment:
`GUINT_TO_POINTER` is an evil thing that subverts the type system. There are two problems in this case: First of all, `time64`s are signed and second they're int64_t. GUINT is an alias for uint32_t, and pointers on 32-bit builds (Windows!) are 32 bits. See the patch on [bug 799104](https://bugs.gnucash.org/show_bug.cgi?id=799104). The user doesn't explain themself well, but I guess they're doing forecasting and have dates past 2038 when time goes over 32 bits.

Reply:
Oops.